README.md

```diff
@@ -417,6 +438,50 @@ cargo_generate_version = ">=0.9.0"
 
 The format for the version requirement is [documented here](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html).
 
+## Conditional template settings.
+
+Using `cargo-generate.toml`, values and some [`Rhai`](https://rhai.rs/book/) syntax, the template author can make certain conditional decisions before expansion of the template.
+
+`include`, `exclude`, `ignore` and `placeholders` can all be used in sections that are only used based upon the value of one or more values, possibly input by the user using the interactive prompt (if the values in question are defined as placeholders in the non-conditional section).
+
+Example:
+
+```toml
+[template]
+cargo_generate_version = ">=0.10.0"
+# ignore = [ "..." ]
+# include = [ "..." ]
+# exclude = [ "..." ]
+
+[placeholders]
+license = { type = "string", prompt = "What license to use?", choices = ["MIT", "Unrestricted"], default = "MIT" }
+
+[conditional.'crate_type == "lib"']
+ignore = [ "src/main.rs" ]
+# include = [ "..." ]
+# exclude = [ "..." ]
+
+[conditional.'crate_type != "lib"'.placeholders]
+hypervisor = { type = "string", prompt = "What hypervisor to use?", choices = ["uhyve", "qemu"], default = "qemu" }
+network_enabled = { type = "bool", prompt = "Want to enable network?", default = true }
+
+[conditional.'license == "MIT"']
```

Comment:
praise: you've chosen pretty nice examples thou

Reply:
Thus, I've chosen to keep the example, but tried to improve upon the explanation :-)